### PR TITLE
Changed scheme to HTTPS

### PR DESCRIPTION
**Pull request by digiltd**
This has fixed several cases of no audio being heard even though playback appears to be working ok. See issues #15 and #14

### CODE REVIEW

**Comment by askvictor** [comment]
I just tried it on my install and it no longer works with this change. Haven't investigated error messages or anything, but it looks like its not quite that simple

**Comment by digiltd** [comment]
Still working for me using:
Logitech Media Server Version: 7.9.0 - 1483028105 @ Thu Dec 29 17:25:39 UTC 2016

Tried a reload of your library?

**Comment by askvictor** [comment]
No, will give that a go. I suspect bumping the plugin version will fix this

**Comment by kettenbach-it** [comment]
I think this needs to be merged, since w/o the Plugin does not work anymore!

**Comment by askvictor** [comment]
I've just had a play with this again and it seems that I don't have the HTTPS.pm module in /usr/share/perl5/Slim/Player/Protocols so the plugin fails to start. I'm running 7.8.1~1463511006 on ubuntu. What version of LMS are you on? Did you do anything to install this module?

Cheers,

**Comment by redlulz** [comment]
it's only on 7.9

**Comment by askvictor** [comment]
Hmm; so moving to HTTPS will break the plugin for versions <7.9; in my case it still works happily on http so I'm hesitant to do this. I'm curious anyone using <7.9 are having this problem.

 Perhaps checking a condition (either a "Use https" setting and/or the LMS version) to determine which module to use would do the job. This might need a trick such as http://stackoverflow.com/questions/3945583/how-can-i-conditionally-use-a-module-in-perl

**Comment by askvictor** [comment]
I'm thinking that I'll start a new branch for 7.9+. This will cause minor annoyances for the package install however.

**Comment by tmancill** [comment]
:+1: for this this patch.  It resolved my issues with LMS 7.9 together with gmusicapi==10.1.2.

**Comment by askvictor** [comment]
@tmancill - as said previously, this patch breaks the plugin on LMS 7.8 and below, so can't be merged as is; options include creating two plugins - one for <= 7.8, one for >= 7.9, or a single plugin that tries to determine this at runtime (I had a look at that a while ago, and it seemed harder than I first imagined).

If anyone wants to become a member of this repo to work on this, please let me know; I'm not in any position to work on any code now, and probably won't until something breaks for me, but I've set up this github organisation so others can contribute.

File: ProtocolHandler.pm [hunk 1]
```diff
@@ -2,7 +2,7 @@ package Plugins::GoogleMusic::ProtocolHandler;
 
 use strict;
 use warnings;
-use base qw(Slim::Player::Protocols::HTTP);
+use base qw(Slim::Player::Protocols::HTTPS);
```
Comment (davewongillies) [inline ProtocolHandler.pm]: Wrapping this in a conditional to check the version I think should work:
```
if ( Slim::Utils::Versions->compareVersions($::VERSION, '7.9.0') < 0 ) {
    eval "use base qw(Slim::Player::Protocols::HTTP)";
} else {
    eval "use base qw(Slim::Player::Protocols::HTTPS)";
}
```
Only tested on 7.9.1.

Comment (davewongillies) [inline ProtocolHandler.pm:5]: Wrapping this in a conditional to check the version should work:
```
if ( Slim::Utils::Versions->compareVersions($::VERSION, '7.9.0') < 0 ) {
    eval "use base qw(Slim::Player::Protocols::HTTP)";
} else {
    eval "use base qw(Slim::Player::Protocols::HTTPS)";
}
```
Only tested on 7.9.1.

**Review by davewongillies — CHANGES_REQUESTED** [review]

File: ProtocolHandler.pm (hunk 1 above)
Comment (davewongillies) [reply ProtocolHandler.pm:5]: I've made my own fork of this repo and commit the change in https://github.com/davewongillies/squeezebox-googlemusic/commit/69529b7a1b420fd0c5ef2fc93e129fef07d8d204

And have also pushed a new version to my fork of the repo here: https://davewongillies.github.io/squeezebox-googlemusic/repository/repo.xml

Comment (digiltd) [reply ProtocolHandler.pm:5]: It is worth submitting a PR here to keep everything in one place?

Pretty sure mine didn't get merged due to it breaking <= 7.8, but as your fix solves this ping @askvictor and see if he will add you to the repo :)

Comment (davewongillies) [reply ProtocolHandler.pm:5]: @digiltd I'm happy to raise another PR with the fix.

Comment (davewongillies) [reply ProtocolHandler.pm:5]: @digiltd do you know if the `streamURL` change mentioned in https://github.com/squeezebox-googlemusic/squeezebox-googlemusic/issues/14 is required as well? I'll test it out on <= 7.8 though (I tested the plugin to make sure it didn't have any errors but didn't test playback).

Comment (digiltd) [reply ProtocolHandler.pm:5]: no idea sorry, i think #14 talks about an either or situation

a working pr is always a good pr :)

Comment (davewongillies) [reply ProtocolHandler.pm:5]: So I've raised https://github.com/squeezebox-googlemusic/squeezebox-googlemusic/pull/25 as a replacement for this PR. If you've got the chance, I'd appreciate if you could test out my changes.

**Comment by davewongillies** [comment]
Closing this PR in favour of #25 . Would love it if I could get some testing of #25. I'll leave it open for a few days and if I don't hear back, I'll merge.

**Pull request closed without merging**